cachito/workers/pkg_managers/general.py

```diff
@@ -157,6 +157,36 @@ def download_binary_file(url, download_path, auth=None, insecure=False, chunk_si
             f.write(chunk)
 
 
+async def async_download_binary_file(
+    session, url, download_dir, tarball_name, auth=None, chunk_size=8192
+):
+    """
+    Download a binary file (such as a TAR archive) from a URL using asyncio.
+
+    :param aiohttp.ClientSession session: Aiohttp interface for making HTTP requests.
+    :param str url: URL for file download
+    :param str download_dir: Path to download file to
+    :param str tarball_name: Name of the file
+    :param aiohttp.BasicAuth auth: Authentication for the URL
+    :param int chunk_size: Chunk size param for Response.content.read()
```

Comment:
We try to use type annotations for all new code, could you move the types from the docstring to the definition?

(same for the other new functions)